### PR TITLE
Use in-memory filesystem for the file sink and source

### PR DESCRIPTION
**Pull request by johanandren**
Should stabilize the tests on the CI-server.

Refs #20726

### CODE REVIEW

**Comment by ktoso** [comment]
Great, LGTM

File: akka-stream-tests/src/test/scala/akka/stream/io/FileSinkSpec.scala [hunk 1]
```diff
@@ -136,7 +141,7 @@ class FileSinkSpec extends StreamSpec(UnboundedMailboxConfig) {
   }
 
   private def targetFile(block: Path ⇒ Unit, create: Boolean = true) {
-    val targetFile = Files.createTempFile("synchronous-file-sink", ".tmp")
+    val targetFile = Files.createTempFile(fs.getPath("/"), "synchronous-file-sink", ".tmp")
```
Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/io/FileSinkSpec.scala]: PS: so that just works like that? Awesome.

**Comment by drewhk** [comment]
wow great! LGTM

File: akka-stream-tests/src/test/scala/akka/stream/io/FileSinkSpec.scala (hunk 1 above)
Comment (johanandren) [inline akka-stream-tests/src/test/scala/akka/stream/io/FileSinkSpec.scala]: Yeah, pretty nice there Java (I have not forgiven you for FileWatch yet though)

**Comment by patriknw** [comment]
LGTM, if we see more trouble with the LevelDB based tests we should perhaps give this a try on them as well

**Pull request merged**